README.md

```diff
@@ -173,6 +173,34 @@ Using the `Parameters` classes is not required, but they can help in a number of
 - Provides a more natural way to construct parameters
 - Facilitates ASP.NET Core model binding (bind an HTML form to a `Parameters` instance)
 
+## Debugging
+
+### View HTTP Requests
+
+Users can audit the HTTP requests and responses being made by the library by setting the `Hooks` property of a `ClientConfiguration` with a set of event handlers. Available handlers include:
+
+- `OnRequestBeforeExecution` - Called before an HTTP request is made. An `RequestBeforeExecutionEventArgs` object is passed to the handler, which contains the `HttpRequestMessage` that will be sent to the server.
```

Comment:
issue: per the design doc and where possible, we should try to keep the language close to `RequestHook`. We can then document that it fires before a request happens. These names are pretty verbose.

Reply:
This is standard with how event/delegate verbiage typically works in .NET.

Reply:
Thanks, I hate it